pkg/model/matches.go

```diff
@@ -0,0 +1,121 @@
+package model
+
+import (
+	"math"
+	"strings"
+
+	"github.com/grafana/tempo/pkg/tempopb"
+	v1common "github.com/grafana/tempo/pkg/tempopb/common/v1"
+	v1 "github.com/grafana/tempo/pkg/tempopb/trace/v1"
+	"github.com/grafana/tempo/pkg/util"
+	"github.com/grafana/tempo/tempodb/search"
+)
+
+const RootSpanNotYetReceivedText = "<root span not yet received>"
+
+// Matches determines if the passed object encoded using dataEncoding matches the tempopb.SearchRequest.
+//  If the object matches the request then a non-nil tempopb.TraceSearchMetaData is returned. Otherwise
+//  nil is returned.
+func Matches(id []byte, obj []byte, dataEncoding string, reqStart, reqEnd uint32, req *tempopb.SearchRequest) (*tempopb.TraceSearchMetadata, error) {
+	start := uint64(math.MaxUint64)
+	end := uint64(0)
```

Comment:
On a first read I was confusing these two parameters with start and end from the search request, maybe consider renaming them to `traceStart` and `traceEnd` to clarify these are timestamps?

Reply:
+1